File: packages/python-sdk/src/urisocial/resources/billing.py
"""Billing and credits resource"""

from typing import TYPE_CHECKING, Optional
from ..types import BillingInfo

if TYPE_CHECKING:
    from ..http_client import HTTPClient


class BillingResource:
    """Resource for billing and credit management"""

    def __init__(self, http: "HTTPClient"):
        self._http = http

    def get_info(self) -> BillingInfo:
        """
        Get current billing information and credits

        Returns:
            Billing info with credits remaining and subscription tier

        Example:
            >>> billing = client.billing.get_info()
            >>> print(f"Credits: {billing['credits_remaining']}")
            >>> print(f"Tier: {billing['subscription_tier']}")
        """
        return self._http.get("/api/v1/billing/info")

    def get_usage(
        self,
        start_date: Optional[str] = None,
        end_date: Optional[str] = None,
    ) -> dict:
        """
        Get usage history

        Args:
            start_date: Optional start date (ISO format)
            end_date: Optional end date (ISO format)

        Returns:
            Usage history with daily breakdown

        Example:
            >>> usage = client.billing.get_usage(
            ...     start_date='2024-01-01',
            ...     end_date='2024-01-31'
            ... )
            >>> print(f"Total API calls: {usage['total_api_calls']}")
        """
        params = {}
        if start_date:
            params["start_date"] = start_date
        if end_date:
            params["end_date"] = end_date

        return self._http.get(
            "/api/v1/billing/usage",
            params=params if params else None,
        )

File: packages/typescript-sdk/src/resources/billing.ts
import { HTTPClient } from '../client/http';

/**
 * Usage information for SDK users
 * Provides visibility into API usage without exposing billing internals
 */
export interface UsageInfo {
  credits_remaining: number;
  api_calls_this_month: number;
  rate_limit_remaining?: number;
  rate_limit_reset?: string;
}

export interface UsageHistory {
  date: string;
  api_calls: number;
  operations: Array<{
    operation_type: string;
    count: number;
  }>;
}

/**
 * Billing/Usage Resource
 *
 * Provides read-only access to usage statistics and credit information.
 * Payment and subscription management should be done through your URI Social dashboard.
 */
export class BillingResource {
  constructor(private http: HTTPClient) {}

  /**
   * Get current usage information
   * Shows credits remaining and API usage statistics
   */
  async getUsageInfo(): Promise<UsageInfo> {
    // This should call a public SDK endpoint that returns usage info
    // without exposing internal billing details
    return this.http.get<UsageInfo>('/api/v1/billing/info');
  }

  /**
   * Get usage history
   * @param startDate - Start date (YYYY-MM-DD)
   * @param endDate - End date (YYYY-MM-DD)
   */
  async getUsageHistory(startDate?: string, endDate?: string): Promise<{
    usage: UsageHistory[];
    total_api_calls: number;
  }> {
    const params = new URLSearchParams();
    if (startDate) params.append('start_date', startDate);
    if (endDate) params.append('end_date', endDate);
    const query = params.toString() ? `?${params.toString()}` : '';
    return this.http.get(`/api/v1/billing/usage${query}`);
  }
}
